### PR TITLE
FIX #235: change RX requirements to a version below 3.0

### LINKED ISSUE #235: RX dependancy conflict for Python 2.7
It seems RX > 3.0 is not anymore compatible with python 2.7, so I get this error on graphene-sqlalchemy build, as mentionned in https://github.com/graphql-python/graphene-sqlalchemy/pull/177#issuecomment-477218178

I prepare a PR to fix it

### PR DESCRIPTION
**Pull request by Nabellaleen**
Fix #235

### CODE REVIEW

**Comment by Nabellaleen** [comment]
Maybe we should apply this fix to older graphql-core versions

**Review by ProjectCheshire — COMMENTED** [review]
Do we want to drop the greater than 1.6.0? I would assume we want gte 1.6.0 and lt 3.0.0

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 sys.path[:] = path_copy
 
-install_requires = ["six>=1.10.0", "promise>=2.1", "rx>=1.6.0"]
+install_requires = ["six>=1.10.0", "promise>=2.1", "rx<3.0.0"]
```
Comment (dan98765) [inline setup.py:24]: Same suggestion as @ProjectCheshire 
`1.6.0 >= rx < 3.0.0`

**Review by dan98765 — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (SmileyChris) [reply setup.py:24]: Using environment markers seems like the correct solution. There's no need to limit RX for people actually using Python 3 is there?
```suggestion
install_requires = [
    "six>=1.10.0",
    "promise>=2.1",
    'rx>=1.6.0,<3;python_version<"3"',
    'rx>=1.6.0;python_version>="3"',
]
```

Comment (ProjectCheshire) [reply setup.py:24]: Possibly out of scope of this PR, but brings up related dependencies amongst all the repos @dan98765 @SmileyChris

https://github.com/graphql-python/graphql-ws/blob/5f2cff5578b52f6c363567e5252d5e5d48e64787/graphql_ws/observable_aiter.py

Uses the extensionmethod that no longer exists in rx => 3, so the user python won't matter there

Note, I believe graphql core and ws next no longer requires rx

Comment (jeremytubbs) [reply setup.py:24]:
```suggestion
install_requires = [
    "six>=1.10.0",
    "promise>=2.1",
    'rx>=1.6.0,<3;python_version<"3.6"',
    'rx>=1.6.0;python_version>="3.6"',
]
```

Comment (Cito) [reply setup.py:24]: I would keep the minimum requirement and make it "rx>=1.6,<3".

Comment (jeremytubbs) [reply setup.py:24]: I just know that this needs to be resolved... rx came out of Beta yesterday and is breaking installations.

Comment (Cito) [reply setup.py:24]: Ok, I will try to publish a new release as a hotfix.

Comment (Cito) [reply setup.py:24]: Hotfix release 2.2.1 is out now. Let me know if there are any problems with this release.

**Comment by Cito** [comment]
Thank you all. This is now fixed in version 2.2.1.

**Pull request closed without merging**